agent/socket.go

```diff
@@ -190,3 +194,23 @@ func (s *Server) extendPod(ctx context.Context, channel, namespace, pod string,
 	}
 	return nil
 }
+
+func updateDeletionTime(ctx context.Context, namespace, pod string, tm time.Time) error {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	po, err := clientset.CoreV1().Pods(namespace).Get(ctx, pod, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	runnerClient := client.NewClient()
```

Comment:
Why don't you reuse the client?

Reply:
Thank you for your question.
I didn't have a clear reason for it.
I modify the server structure to have a runnerClient.